service/src/main/java/com/epam/esm/validator/impl/CertificateValidator.java

```diff
@@ -0,0 +1,63 @@
+package com.epam.esm.validator.impl;
+
+import com.epam.esm.entity.Certificate;
+import com.epam.esm.exception.ValidatorException;
+import com.epam.esm.validator.Validator;
+import org.springframework.stereotype.Service;
+
+import java.math.BigDecimal;
+
+@Service
+public class CertificateValidator implements Validator<Certificate> {
+
+    private final int MIN_NAME_SIZE = 3;
+    private final int MAX_NAME_SIZE = 30;
+    private final int MAX_DESCRIPTION_SIZE = 150;
+
+    @Override
+    public void validate(Certificate certificate) throws ValidatorException {
+        if (certificate == null) {
+            throw new ValidatorException("Certificate should be not null");
+        }
+        validateName(certificate.getName());
+        validateDescription(certificate.getDescription());
+        validateDuration(certificate.getDuration());
+        validatePrice(certificate.getPrice());
```

Comment:
This class will throw an error as soon as it encounters the first validation error.
But what happens if the request contains several errors, for example, there is no name and description?
It would be great if the client received all error messages at once.